### PR TITLE
Fix #6650

### LINKED ISSUE #6650: [BUG][4.3.314][Calendar] erroneous display of tasks/events in view Calendar
#### Issue

- The end date is displayed until a day before indicated
image

- (Other minor bug) when you select "All day" is saved in the database, start time 8:00 and end time 8:00, maybe you can give a problem when you select "Fill the start date based on the first available date"
firefox_2018-04-25_10-16-55

#### How to trigger the error

https://puu.sh/Aa1Si/d022fd5522.mp4

#### Your Environment

* YetiForce Version used: 4.3.314
* Browser name and version: Firefox 59
* Environment name and version: gitdeveloper.yetiforce.com
* Operating System and version: gitdeveloper.yetiforce.com

### PR DESCRIPTION
**Pull request by WalterLuis**
## Description

#6650 

## Changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code is written according to the guidelines found [here] (https://yetiforce.com/en/knowledge-base/documentation/developer-documentation/item/how-to-create-php-files).
- [X] I have signed the Contributor Licence Agreement between me and YetiForce.

### CODE REVIEW

File: modules/Calendar/actions/Save.php [hunk 1]
```diff
@@ -65,26 +65,32 @@ public function saveRecord(\App\Request $request)
 	protected function getRecordModelFromRequest(\App\Request $request)
 	{
 		$recordModel = parent::getRecordModelFromRequest($request);
+		$user = Users_Record_Model::getCurrentUserModel();
+		$allDay = $request->get('allday');
+		if ($allDay) {
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/actions/Save.php]: > $allDay = $request->get('allday');
> if ($allDay) {

change to

> if ($request->getBoolean('allday')) {

File: modules/Calendar/actions/SaveAjax.php [hunk 2]
```diff
@@ -15,12 +15,11 @@ public function process(\App\Request $request)
 	{
 		$user = Users_Record_Model::getCurrentUserModel();
 		$allDay = $request->get('allday');
-		if ('on' === $allDay) {
+		if ($allDay) {
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/actions/SaveAjax.php]: > $allDay = $request->get('allday');
> if ($allDay) {

change to

> if ($request->getBoolean('allday')) {

File: modules/Calendar/actions/Save.php [hunk 3]
```diff
@@ -65,26 +65,32 @@ public function saveRecord(\App\Request $request)
 	protected function getRecordModelFromRequest(\App\Request $request)
 	{
 		$recordModel = parent::getRecordModelFromRequest($request);
+		$user = Users_Record_Model::getCurrentUserModel();
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/actions/Save.php]: > $user = Users_Record_Model::getCurrentUserModel();

change to

> $user = \App\User::getCurrentUserModel();

File: modules/Calendar/actions/Save.php [hunk 4]
```diff
@@ -65,26 +65,32 @@ public function saveRecord(\App\Request $request)
...
+		$allDay = $request->get('allday');
+		if ($allDay) {
+			$request->set('time_start', $user->get('start_hour'));
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/actions/Save.php]: > $request->set('time_start', $user->get('start_hour'));

change to

> $request->set('time_start', $user->getDetail('start_hour'));

File: modules/Calendar/actions/Save.php [hunk 5]
```diff
@@ -65,26 +65,32 @@ public function saveRecord(\App\Request $request)
...
+			$request->set('time_end', $user->get('end_hour'));
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/actions/Save.php]: > $request->set('time_end', $user->get('end_hour'));

change to

> $request->set('time_end', $user->getDetail('end_hour'));

**Comment by paula-w** [comment]
Thanks again @WalterLuis ! you're the best! 😃

**Comment by WalterLuis** [comment]
thanks to you again @paula-w ❤️

**Pull request merged**